src/core/server/saved_objects/mappings/utils/prose.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ *  Converts an array of items into a sentence-ready string.
+ *
+ *  @param {string[]} list
+ *  @param {Object} [options={}]
+ *  @property {Boolean} [options.inclusive=true] Creates an inclusive list using "and"
+ *                                               when `true` (default), otherwise uses "or"
+ *  @return {String}
+ */
+export function formatListAsProse(list: string[], options = { inclusive: true }) {
```

Comment:
I had this idea that we'd move utils slowly over to typescript. Should this not be done here?

Reply:
If you want to just convert the other util into typescript and reference it directly (rather than importing the whole `utils` module) I'd be fine with that too. But I also don't mind duplicating it here and keeping dependencies local.

Reply:
Ok, then leaving as is.